### PR TITLE
COMMERCE-12676 Add automation CannotAddToCartOptionWhenStockIsZeroAndBackOrderIsDisabled

### PR DESCRIPTION
**Pull request by jaydawu**
https://liferay.atlassian.net/browse/COMMERCE-12678

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase [hunk 1]
```diff
@@ -1774,6 +1774,105 @@ definition {
...
+			CommerceSKUs.generateAllSkuCombinations();
+		}
+
+		task ("And the inventory of other generated SKUs is configured besides SKU:6 which has 0 stock") {
+			for (var sku : list "112,48,24,12") {
+				CommerceNavigator.gotoEntry(entryName = ${sku});
+
+				CommerceEntry.editSkuPricingAndInventory(italyQuantity = 10);
+			}
+		}
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductDetails.testcase:1823]: Can you use JSON macros instead? Thanks!

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]

**Comment by lorenzolapescara** [comment]
Manually forwarded.

**Comment by lorenzolapescara** [comment]
ci:close

**Pull request closed without merging**